src/main/java/org/broadinstitute/hellbender/tools/reference/CompareReferences.java

```diff
@@ -204,6 +256,193 @@ private void displayBySequenceName(Set<ReferenceSequenceTable.TableRow> rows, St
+                throw new UserException("");
+            }
+        }
+        else{
+            logger.info("No mismatching sequences found.");
+        }
+    }
+
+    private void runFindSNPS(ReferencePair refPair, ReferenceSequenceTable table) {
+        if(refPair.getAnalysis().contains(ReferencePair.Status.DIFFER_IN_SEQUENCE)) {
+            TableColumnCollection columns = new TableColumnCollection(Arrays.asList("Sequence Name", "Position", refPair.getRef1AsString(), refPair.getRef2AsString()));
+            File snpsOutput = new File(baseComparisonOutputDirectory.toPath().toString(), String.format("%s_%s_snps.tsv", refPair.getRef1AsString(), refPair.getRef2AsString()));
+
+            try(FindSNPsOnlyTableWriter writer = new FindSNPsOnlyTableWriter(snpsOutput.toPath(), columns); final ReferenceDataSource source1 = ReferenceDataSource.of(refPair.getRef1().toPath(), true);
+                final ReferenceDataSource source2 = ReferenceDataSource.of(refPair.getRef2().toPath(), true)){
+                writer.writeHeaderIfApplies();
+
+                // find the mismatch sequence
+                for (String sequenceName : table.getAllSequenceNames()) {
+                    Set<ReferenceSequenceTable.TableRow> rows = table.queryBySequenceName(sequenceName);
+                    if (rows.size() == 2) {
+                        // if the lengths of the 2 sequences aren't equal, error - can't compare different sequence lengths, probably indel need alignment
+                        ReferenceSequenceTable.TableRow[] rowArray = rows.toArray(new ReferenceSequenceTable.TableRow[0]);
+                        if (rowArray[0].getLength() != rowArray[1].getLength()) {
+                            logger.warn("Sequence lengths are not equal and can't be compared. Consider running in FULL_ALIGNMENT mode.");
+                            continue;
+                        }
+
+                        int sequenceLength = rowArray[0].getLength();
+                        Iterator<Byte> ref1BaseIterator = source1.query(new SimpleInterval(sequenceName, 1, sequenceLength));
+                        Iterator<Byte> ref2BaseIterator = source2.query(new SimpleInterval(sequenceName, 1, sequenceLength));
+                        int position = 0;
+
+                        // find SNPS and generate SNPRecords
+                        while (ref1BaseIterator.hasNext()) {
+                            position++;
+                            Byte ref1Allele = ref1BaseIterator.next();
+                            Byte ref2Allele = ref2BaseIterator.next();
+                            if (!ref1Allele.equals(ref2Allele)) {
+                                SNPRecord record = new SNPRecord(sequenceName, position, new String(new byte[]{ref1Allele}), new String(new byte[]{ref2Allele}), refPair.getRef1AsString(), refPair.getRef2AsString());
+                                writer.writeRecord(record);
+                            }
+                        }
```

Comment:
add a check at the end that both are finished when you kick out and throw an exception otherwise.

Reply:
do i need this in addition to the check earlier in the loop that the sequence lengths for the current sequence are equal? if i want one specifically on these iterators, would that be a user exception? should be same sequence length as is being used in the first check

Reply:
@orlicohen This is an error that should never happen, since you've already checked that the sequence lengths are the same. If it does happen that one iterator finishes before the other, something has gone horribly wrong internally. You'd probably want to throw a `ShouldNeverReachHereException` in that case, since we never expect it to happen in practice.